kratos/processes/from_json_check_result_process.h

```diff
@@ -276,26 +277,24 @@ class KRATOS_API(KRATOS_CORE) FromJSONCheckResultProcess
         for (auto& p_var_double : mpNodalVariableDoubleList) {
             const auto& r_var_database = r_node_database.GetVariableData(*p_var_double);
 
-            #pragma omp parallel for reduction(+:check_counter)
-            for (int i = 0; i < static_cast<int>(r_nodes_array.size()); ++i) {
-                auto it_node = it_node_begin + i;
-
+            check_counter = IndexPartition<std::size_t>(r_nodes_array.size()).for_each<SumReduction<IndexType>>([&] (std::size_t Index){
+                auto it_node = it_node_begin + Index;
                 const double result = GetValue<THistorical>(it_node, p_var_double);
-                const double reference = r_var_database.GetValue(i, time);
+                const double reference = r_var_database.GetValue(Index, time);
                 if (!CheckValues(result, reference)) {
                     FailMessage(it_node->Id(), "Node", result, reference, p_var_double->Name());
                     check_counter += 1;
```

Comment:
can you please declare this variable inside the loop?
Otherwise I am not sure if it gets accessed from multiple threads or not